nexus/db-queries/src/db/datastore/physical_disk.rs

```diff
@@ -278,7 +278,28 @@ impl DataStore {
             .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))
     }
 
+    /// Decommissions all expunged disks.
+    pub async fn physical_disk_decommission_all_expunged(
+        &self,
+        opctx: &OpContext,
+    ) -> Result<(), Error> {
+        opctx.authorize(authz::Action::Modify, &authz::FLEET).await?;
+        use db::schema::physical_disk::dsl;
+
+        let conn = &*self.pool_connection_authorized(&opctx).await?;
+        diesel::update(dsl::physical_disk)
+            .filter(dsl::time_deleted.is_null())
+            .physical_disk_filter(DiskFilter::ExpungedButActive)
+            .set(dsl::disk_state.eq(PhysicalDiskState::Decommissioned))
+            .execute_async(conn)
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))?;
+        Ok(())
+    }
+
     /// Deletes a disk from the database.
+    // TODO: This probably wants to take a "UUID" as an argument, only operate
+    // on expunged disks?
```

Comment:
follow up issue?

Reply:
This was small enough (used by tests only) that I just patched it in this PR

Reply:
:+1: